### PR TITLE
Spring hw 3 solution

### PR DESCRIPTION
**Pull request by AnatolyNz**

### CODE REVIEW

File: src/main/java/mate/academy/WebDtoApplication.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package mate.academy;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class WebDtoApplication {
```
Comment (okuzan) [inline src/main/java/mate/academy/WebDtoApplication.java]:
```suggestion
public class BookStoreApplication {
```
unusual name for the main class

File: mvnw [hunk 2]
```diff
@@ -161,7 +161,7 @@ find_maven_basedir() {
 concat_lines() {
   if [ -f "$1" ]; then
     # Remove \r in case we run on Windows within Git Bash
-    # and check out the repository with auto CRLF management
+    # and check out the mate.academy.repository with auto CRLF management
```
Comment (okuzan) [inline mvnw:164]: mvn files shouldn't be committed to the repo

File: pom.xml [hunk 3]
```diff
@@ -5,60 +5,89 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>3.2.0</version>
-        <relativePath/>
+        <version>3.1.5</version>
+        <relativePath/> <!-- lookup parent from repository -->
```
Comment (okuzan) [inline pom.xml]:
```suggestion
        <relativePath/> 
```

File: src/main/java/mate/academy/repository/BookDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+
+    @Override
+    public Optional<Book> getBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT book FROM Book book WHERE book.id = :id", Book.class)
+                    .setParameter("id", id)
+                    .uniqueResultOptional();
+        }
```
Comment (okuzan) [inline src/main/java/mate/academy/repository/BookDaoImpl.java:61]:
```suggestion
        } catch (Exception e) {
        throw new RuntimeException("Can't get book by id from DB: " + id, e);
    }
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List getAll() {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/controller/BookController.java:22]: It is bad practice, you need to set the Type
```suggestion
    public List<BookDto> getAll() {
```

**Review by Ivan95kos — APPROVED** [review]
Great job, with a minor comment

**Pull request closed without merging**